File: src/main/java/net/slipp/web/QnAPageableHelper.java
package net.slipp.web;

import org.springframework.data.domain.PageRequest;
import org.springframework.data.domain.Pageable;
import org.springframework.data.domain.Sort;
import org.springframework.data.domain.Sort.Direction;

public class QnAPageableHelper {
	public static final int DEFAULT_PAGE_NO = 1;
	
	public static Pageable createPageableByQuestionUpdatedDate(Integer currentPage, Integer pageSize) {
		Sort sort = new Sort(Direction.DESC, "updatedDate");
		return new PageRequest(currentPage - 1, pageSize, sort);
	}
	
	public static Pageable createPageableByAnswerCreatedDate(Integer currentPage, Integer pageSize) {
		Sort sort = new Sort(Direction.DESC, "createdDate");
		return new PageRequest(currentPage - 1, pageSize, sort);
	}
	
	public static Integer revisedPage(Integer page) {
		if (page == null) {
			page = DEFAULT_PAGE_NO;
		}
		return page;
	}
}

File: src/main/scala/net/slipp/web/QnAPageableHelper.scala
package net.slipp.web

import org.springframework.data.domain.Sort
import org.springframework.data.domain.PageRequest
import org.springframework.data.domain.Sort.Direction

object QnAPageableHelper {
  val DefaultPageNo = 1

  def createPageableByQuestionUpdatedDate(currentPage: Integer, pageSize: Integer) = {
    createPageable(currentPage, pageSize, "updatedDate")
  }

  def createPageableByAnswer(currentPage: Integer, pageSize: Integer) = {
    new PageRequest(revisedPage(currentPage, DefaultPageNo) -1, pageSize);
  }
  
  def createPageableById(currentPage: Integer, pageSize: Integer) = {
    createPageable(currentPage, pageSize, "id")
  }
  
  def createPageableTagId(currentPage: Integer, pageSize: Integer) = {
    createPageable(currentPage, pageSize, "tagId")
  }  
  
  def createPageable(currentPage: Integer, pageSize: Integer, sortField: String) = {
    val sort = new Sort(Direction.DESC, sortField)
    new PageRequest(revisedPage(currentPage, DefaultPageNo) - 1, pageSize, sort)
  }

  def revisedPage(page: Integer): Integer = revisedPage(page, DefaultPageNo)
  
  def revisedPage(page: Integer, defaultPageNo: Integer) = if (page == null) defaultPageNo else page
}